source/PlayerInfo.cpp

```diff
@@ -2698,11 +2771,131 @@ void PlayerInfo::CreateMissions()
 		}
 	}
 
-	// Sort missions on the job board alphabetically.
-	availableJobs.sort([](const Mission &lhs, const Mission &rhs)
+	// Sort missions on the job board.
+	SortAvailable();
+}
+
+
+
+void PlayerInfo::SortAvailable()
+{
+	// Destinations: planets OR system. Only counting them, so the type doesn't matter.
+	set<const void*> destinations;
+	if(availableSortType == CONVENIENT)
+	{
+		for(const Mission &mission : Missions())
+		{
+			if(mission.IsVisible())
+			{
+				destinations.insert(mission.Destination());
+				destinations.insert(mission.Destination()->GetSystem());
+
+				for(const Planet* stopover : mission.Stopovers())
+				{
+					destinations.insert(stopover);
+					destinations.insert(stopover->GetSystem());
+				}
+
+				for(const System* waypoint : mission.Waypoints())
+					destinations.insert(waypoint);
+			}
+		}
+	}
+	availableJobs.sort([&](const Mission &lhs, const Mission &rhs)
 	{
-		return lhs.Name() < rhs.Name();
+		// First, separate rush orders if wanted
```

Comment:
I'm not sure what separate rush orders mean, this comment could be more helpful
Rush refers to the missions with a deadline I guess? So in fact this should be named sortMissionByDeadlines?

Reply:
These are called "separate" instead of "sort" for a few reasons:

They are not the primary sorting method - that's alphabetical, or by pay, etc.

They "sort" by a binary value, instead of sorting a range of possible values: it's more separating true from false, than really sorting a list. 

They are also physically separated when displayed, with an 8-pixel gap between separated missions.

It could be named deadline. I guess all the code calls it though and it's just missions that are named "rush delivery"